microsoft/testsuites/core/timesync.py

```diff
@@ -1,9 +1,29 @@
 from pathlib import PurePosixPath
+from typing import List, Optional, Union
 
 from assertpy import assert_that
+from retry import retry
 
-from lisa import Environment, Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
-from lisa.tools import Cat, Dmesg
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.tools import Cat, Dmesg, Lscpu
+from lisa.tools.lscpu import CpuType
+
+
+@retry(tries=150, delay=0.5)
+def _check_value(
+    node: Node,
+    path: str,
+    expected_value: Optional[Union[str, List[str]]],
+) -> str:
+    cat = node.tools[Cat]
+    result = cat.run(path, force_run=True)
+    if isinstance(expected_value, List):
```

Comment:
Does it work? the `List` may need to be `list`.

Reply:
The retry works only if there are exception raised.

Reply:
does return will exist the retry?

Reply:
If no exception, it won't retry.